diff --git a/crypto/bn/asm/armv4-gf2m.s b/crypto/bn/asm/armv4-gf2m.s
index 2c209e1..9a07533 100644
--- a/crypto/bn/asm/armv4-gf2m.s
+++ b/crypto/bn/asm/armv4-gf2m.s
@@ -1,5 +1,8 @@
 #include "arm_arch.h"
 
+@ define __ARM_ARCH__<7 to workaround -Bdynamic build issue
+#define __ARM_ARCH__ 6
+
 .text
 .code	32
 
diff --git a/crypto/sha/asm/sha512-armv4.s b/crypto/sha/asm/sha512-armv4.s
index 5730192..9fd535f 100644
--- a/crypto/sha/asm/sha512-armv4.s
+++ b/crypto/sha/asm/sha512-armv4.s
@@ -1,4 +1,8 @@
 #include "arm_arch.h"
+
+@ define __ARM_ARCH__<7 to workaround -Bdynamic build issue
+#define __ARM_ARCH__ 6
+
 #ifdef __ARMEL__
 # define LO 0
 # define HI 4
